docs/overview.rst

```diff
@@ -1,380 +1,306 @@
+- :attr:`Contract.bytecode_runtime <web3.contract.Contract.bytecode_runtime>`
+- :attr:`Contract.functions <web3.contract.Contract.functions>`
+- :attr:`Contract.events <web3.contract.Contract.events>`
+- :attr:`Contract.fallback <web3.contract.Contract.fallback.call>`
+- :meth:`Contract.constructor() <web3.contract.Contract.constructor>`
+- :meth:`Contract.encodeABI() <web3.contract.Contract.encodeABI>`
+- :attr:`web3.contract.ContractFunction <web3.contract.ContractFunction>`
+- :attr:`web3.contract.ContractEvents <web3.contract.ContractEvents>`
 
-    .. WARNING::
-      This method has been deprecated for :meth:`~Web3.solidityKeccak`
 
+Logs and Filters
+----------------
 
-    Returns the sha3 as it would be computed by the solidity ``sha3`` function
-    on the provided ``value`` and ``abi_types``.  The ``abi_types`` value
-    should be a list of solidity type strings which correspond to each of the
-    provided values.
+If you want to react to new blocks being mined or specific events being emitted by
+a contract, you can leverage Web3.py filters.
 
+.. code-block:: python
 
-    .. code-block:: python
+   # Use case: filter for new blocks
+   >>> new_filter = web3.eth.filter('latest')
 
-        >>> Web3.soliditySha3(['bool'], [True])
-        HexBytes("0x5fe7f977e71dba2ea1a68e21057beebb9be2ac30c6410aa38d4f3fbe41dcffd2")
+   # Use case: filter for contract event "MyEvent"
+   >>> new_filter = deployed_contract.events.MyEvent.createFilter(fromBlock='latest')
 
-        >>> Web3.soliditySha3(['uint8', 'uint8', 'uint8'], [97, 98, 99])
-        HexBytes("0x4e03657aea45a94fc7d47ba826c8d667c0d1e6e33a64a036ec44f58fa12d6c45")
+   # retrieve filter results:
+   >>> new_filter.get_all_entries()
+   >>> new_filter.get_new_entries()
 
-        >>> Web3.soliditySha3(['uint8[]'], [[97, 98, 99]])
-        HexBytes("0x233002c671295529bcc50b76a2ef2b0de2dac2d93945fca745255de1a9e4017e")
+More complex patterns for creating filters and polling for logs can be found in the
+:ref:`Filtering <filtering>` documentation.
 
-        >>> Web3.soliditySha3(['address'], ["0x49EdDD3769c0712032808D86597B84ac5c2F5614"])
-        HexBytes("0x2ff37b5607484cd4eecf6d13292e22bd6e5401eaffcc07e279583bc742c68882")
 
-        >>> Web3.soliditySha3(['address'], ["ethereumfoundation.eth"])
-        HexBytes("0x913c99ea930c78868f1535d34cd705ab85929b2eaaf70fcd09677ecd6e5d75e9")
+API
+^^^
 
-Check Encodability
-~~~~~~~~~~~~~~~~~~~~
+- :meth:`web3.eth.filter() <web3.eth.Eth.filter>`
+- :meth:`web3.eth.getFilterChanges() <web3.eth.Eth.getFilterChanges>`
+- :meth:`web3.eth.getFilterLogs() <web3.eth.Eth.getFilterLogs>`
+- :meth:`web3.eth.uninstallFilter() <web3.eth.Eth.uninstallFilter>`
+- :meth:`web3.eth.getLogs() <web3.eth.Eth.getLogs>`
+- :meth:`Contract.events.your_event_name.createFilter() <web3.contract.Contract.events.your_event_name.createFilter>`
+- :meth:`Contract.events.your_event_name.build_filter() <web3.contract.Contract.events.your_event_name.build_filter>`
```

Comment:
Maybe also add `your_event.processReceipt`, `your_event.processLog`, `your_event.getLogs`

Reply:
Looks like a small ticket worth of work to add a `getLogs` method to the events documentation, plus sort out links for `processLog` and `processReceipt`. The latter two get lumped under the same top-level link without links of their own ([ContractEvents.myEvent](https://web3py.readthedocs.io/en/stable/contracts.html#web3.contract.ContractEvents.myEvent)). Will look to sort that out in a subsequent PR, so I can start this beta release.